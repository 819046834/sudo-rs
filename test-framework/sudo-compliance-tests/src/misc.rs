use sudo_test::{Command, Env};

use crate::{Result, SUDOERS_ALL_ALL_NOPASSWD};

#[test]
fn user_not_in_passwd_database_cannot_use_sudo() -> Result<()> {
    let env = Env(SUDOERS_ALL_ALL_NOPASSWD).build()?;

    let output = Command::new("sudo")
        .arg("true")
        .as_user_id(1000)
        .exec(&env)?;

    assert!(!output.status().success());
    assert_eq!(Some(1), output.status().code());

    if sudo_test::is_original_sudo() {
        assert_contains!(
            output.stderr(),
            "sudo: you do not exist in the passwd database"
        );
    }

    Ok(())
}
