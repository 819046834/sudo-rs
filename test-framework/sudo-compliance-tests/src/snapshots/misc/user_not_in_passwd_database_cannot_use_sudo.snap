---
source: sudo-compliance-tests/src/misc.rs
expression: output.stderr()
---
sudo: you do not exist in the passwd database
